core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -151,6 +151,7 @@ class HistoryServer(
   /** Bind to the HTTP server behind this web interface. */
   override def bind() {
     super.bind()
+    provider.start()
```

Comment:
I'd rather place `provider.start()` after `server.bind()` in `main()`, not here. It feels really odd to start provider when calling bind.

Reply:
That makes sense. Done.